### ISSUE TITLE
Prevent page break in the middle of tables in PDF

### ISSUE THREAD

**Issue by Seb33300** [PrestaShop/PrestaShop#28406]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When a PDF is generated (invoices, delivery slips, etc...), if the document content is too long (if the order has too many items), the information can be too long and the PDF is rendered in 2 pages.

The problem is that the page break is not always happening at the good place.
It can be cut in the middle of a table.

### Expected behavior

IMO, the block under the product list (totals, shipping/payment methods) should never be split on 2 pages.
If the page break happens at this position, the entire block should be moved to the next page.

This can be achieved using the `nobr="true"` on tables with TCPDF.
See https://exchangetuts.com/tcpdf-html-table-and-page-breaks-1639689974714064

### Steps to reproduce

1. Enable invoice feature
2. Make a new order with about 12 products
3. Confirm the payment has been received to generate the order
4. Open the order and see the invoice PDF

### PrestaShop version(s) where the bug happened

1.7.8.5

**Comment by hibatallahAouadni** [comment]
Hello @Seb33300 

I didn't reproduce the issue with PS 1785, see the screen record below:

![IN000007](https://user-images.githubusercontent.com/23308427/166654262-c1cd2931-c1bf-43ef-8b73-e30f0e1455d5.gif)

Please check and feedback.
Thanks!

**Comment by Seb33300** [comment]
@hibatallahAouadni you put too many items to your order. 

You should try with 12 items.

**Comment by hibatallahAouadni** [comment]
Hello @Seb33300 

Do you mean that the Total table is split over two pages?

![image](https://user-images.githubusercontent.com/23308427/166703018-dfe53e15-47d2-4c3e-9a81-03e548a21ef1.png)

Waiting for your feedback.
Thanks!

**Comment by Seb33300** [comment]
Yes, I think it will be better to never split this table and move it to the second page.

**Comment by hibatallahAouadni** [comment]
Hello @Seb33300 

Well, let's ask the UX-team :wink: 
Ping @PrestaShop/ux-team & @NeOMakinG wdyt about this issue? should we consider it as a bug and fix it (never split the total table) or this is the expected behavior?

Thanks in advance :pray:

**Comment by Seb33300** [comment]
If the UX team agree with this improvement, I can try to work on a pull request to prevent this entire block to be cut and always move it to the next page.

![Screenshot_20220505-095522_Firefox.jpg](https://user-images.githubusercontent.com/915273/166857728-df7db681-320c-4305-8a3b-031aa9e1c5b6.jpg)

**Comment by TristanLDD** [comment]
Hi @Seb33300 thank you for your feedback

I agree with the fact that this block should't be splitted, however it would be better if the block could appear on the first page when the list of products is short enough, and be moved to next page only when the list is too long and makes it split. Maybe you already took that in account?  
Thank you

**Comment by Seb33300** [comment]
@TristanLDD Yes, when I said `always move it to the next page`, I mean, in the case there is not enough space on 1st page.

**Comment by hibatallahAouadni** [comment]
Hello @Seb33300 

Thanks @TristanLDD for your answer and approve :pray: 
> however it would be better if the block could appear on the first page when the list of products is short enough, and be moved to next page only when the list is too long and makes it split.

We have a GO :tada: you can work on it :muscle: 
Waiting for your PR :rocket: 

Thanks!
